OpenCS refactoring

Flamefire opened the pull request:
In order to facilitate the IMS feature fixes (see #40 ) and related bug fixes this is a major cleanup/overhaul of (mainly) OpenCustomisationSelector replacing artifacts from the previous automated work on it.

It should greatly improve readability fixing a few potential and actual issues on the way.    
This shouldn't introduce noticeable changes to behavior.

One of the larger, possibly suspicious looking changes is this change(s):
```
- public void run() {
+ public synchronized void run() {
-      synchronized (new Object()) {
            // code
-      }
  }
```

But as can be seen in this reduced example it should be clearly correct.

@antoniu200 I would appreciate if you could have a look on this refactoring. It might help checking the individual commits instead of the full diff, in case anything is unclear. I did just some work splitting this work into as small as possible commits

The next step is fixing the issue identified in https://github.com/Flamefire/android_device_sony_yoshino-common/pull/40#discussion_r1493822152 After that all we should need is https://github.com/Flamefire/android_device_sony_yoshino-common/pull/40/commits/5ead3202cbf15a74f07ac20b5ee55a6ce98a5127 but I'll also have a look at the other commits (in the other PRs)

Flamefire committed:
OpenCS: Remove MapleStarter

Change-Id: I10bcb10672ce32ee8395dc10c572043e39e0f211

Flamefire committed:
OpenCS: Simplify `TAG` definitions

Use the name directly as a string to make the TAG clear.

Change-Id: Iefbc08cb716ff6a9e1f2387608751513872cf991

Flamefire committed:
OpenCS: Use `tm` for `TelephonyManager` instances

For brevity and consistency

Change-Id: I48ad4094bc4ec1a70dc65ed2b4bd958016eae575

Flamefire committed:
OpenCS: Move `SUBSCRIPTION` to other constants

Change-Id: I8179f6823ecea5b116cffc46baf43b3658521940

Flamefire committed:
OpenCS: Make button labels translatable

Change-Id: I382e4e61e1aac2c454ec9c515df4bad48ee7e11e

Flamefire committed:
XPeriaParts: Make button label translatable

Change-Id: If92afa90943fe793bf10b49dc22a43d9b3906761

Flamefire committed:
OpenCS|XPeriaParts: Make preference broadcast easier to understand

Use named constants for values instead of magic numbers that need to be
in sync.

Change-Id: Ib4332b4e1863d477a6372afac550dbe819774e6f

Flamefire committed:
OpenCS: Refactor variable names and constants

Replace auto-generated variable names by correct/meaningful ones and
replace magic numbers & string constants by their correct references.

Change-Id: I86399455caf99e9b4bb16439b420f2e82f40a101

Flamefire committed:
OpenCS: Consistently prefix private members with `m`

Avoids the need to use `this.` in almost all cases which can hence be
removed.
It also improves readability as the difference between parameters and
members is clear now that this is consistent.

Change-Id: I63109c46d2c8f160c1124c435a8f9a9c7f8bc515

Flamefire committed:
OpenCS: Factor out `hasSignal` to `CommonUtil`

The same operation is used in multiple places and at some misses
required checks. Factor out into a well-named function and do it once.

Change-Id: I8063d21a3da8084824630e249f553102ce1f65c5

Flamefire committed:
OpenCS: Simplify AirplaneModeObserver

Check the URI in the observer and notify only if that has changed as can
be expected at the call site.
Correct the URI check to use `equals` while doing that.
Also make the relevant members in `NetworkSwitcher` final.

Change-Id: Iceab264f5f4658c0c84adcdefa12e5d2563667b7

Flamefire committed:
OpenCS: Fix use of `synchronized`

This should be at the method signature.
For a static method this is equivalent to `synchronized(klass.class)`
For the observer classes the `synchronized(new Object)` was a mistake as
it synchronizes on a new object on each invocation, i.e. never.

Change-Id: I0aaf03f769da2bbf50db33dbfc039e043191a68e

Flamefire committed:
OpenCS: Factor out `getSimSlotIndex` & `getSubID` to `CommonUtil`

Avoid repeated use of the same logic in different places especially due
to the use of the Settings constant.

Change-Id: I0a4f8dfe771beda36c45b963669b97bf8a6f2455

Flamefire committed:
OpenCS: Remove resource java file

Not used and gets auto-generated during build of the source XMLs

Change-Id: Id82d3653f6ff2a98a970aace7c1ef31b6c9647a0

Flamefire committed:
OpenCS: Replace magic number in `addFlags` by Intent-flags

Do the same in CustomizationSelectorService as in UserPresentReceiver
and remove the comment (it is in Git history anyway)

Change-Id: If4aed78c17f2efcd54e96574977c8f22dfb8e7f5

Flamefire committed:
OpenCS: Replace the `mRegistered` member of the observers by `mListener`-checks

We don't need to check `mRegistered` is set to True when `mListener` is
non-null, hence we don't need that additional field.
Also make sure the re-enque in the Runnable stops when it has been
unregistered.

Change-Id: I8d0775ada3792386e997c8692bbee0b2a789fc82

antoniu200 commented on hardware/OpenCustomizationSelector/src/com/sonymobile/customizationselector/CustomizationSelectorService.java:
```diff
@@ -10,47 +10,46 @@
+                if (isUserSetupComplete(context)) {
+                    CSLog.d(TAG, "evaluateCarrierBundle - Need to reboot, starting dialog.");
+                    context.startActivity(new Intent(Intent.ACTION_MAIN, null).addCategory(Intent.CATEGORY_HOME)
+                        .addFlags(Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED | Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK));
```
There is possibly an unneeded line break here, very similar to what I saw and fixed previously in [PR 39](https://github.com/Flamefire/android_device_sony_yoshino-common/pull/39/files#diff-179d09fcdc44db77e283449d7db690744bb735fdf6f79f6ab6f85cdb6010d128).
Does this code work?

Flamefire replied:
Searching through framework code and GitHub the this is as intented (how punny...). However here it is inconsistent: `addCategory` should have been on a new line already.    
Also the preferred style seems to be to create an `Intent` variable first, then pass that to `startActivity`. Done in ca8df4d and related 13cb0e6

antoniu200 commented on hardware/OpenCustomizationSelector/src/com/sonymobile/customizationselector/ModemSwitcherActivity.java:
```diff
@@ -103,15 +103,15 @@ public void onCreate(Bundle bundle) {
                     for (String modemName : modemList) {
                         if (modemName.toLowerCase(Locale.getDefault())
-                                .contains(charSequence.toString().toLowerCase(Locale.getDefault()))) {
+                                .contains(s.toString().toLowerCase(Locale.getDefault()))) {
```
Same situation as above. Possibly unwanted newline.
Does this code work?

Flamefire replied:
Yes it does as it did before this change. But looking at this it can be further simplified removing the need for the newline (which is for readability)

antoniu200 commented on hardware/OpenCustomizationSelector/src/com/sonymobile/customizationselector/NS/NSBootReceiver.java:
```diff
@@ -1,27 +1,27 @@
     public void onReceive(Context context, Intent intent) {
         if (context == null) {
-            CSLog.d("NSBootReceiver", "Error: Context was null");
+            CSLog.d(TAG, "Error: Context was null");
```
Shouldn't this be `CSLog.e`, for `Error`?
It is currently `CSLog.d`, for `Debug`, but is printing `Error: Context was null`.

Flamefire replied:
Makes sense, resolved in 6424d1d6 unifying also similar things

antoniu200 commented:
Yeah, overall nice changes. Makes the code much clearer, especially in some fixed int cases, like `addFlags(270565376)`.
Really good work!

Flamefire committed:
OpenCS: Fix error logs

Consistently use `.e` instead of `.d` and unify messages

Change-Id: Ic6006e658fa1f26d9c87fa8d09fd111d86783c71

Flamefire committed:
OpenCS: Simplify code using `Locale.getDefault()`

Change-Id: I5c8ea708af45a9ac1ce5c1718bf75b22c6fa4356

Flamefire committed:
CS: Remove duplication in logfile path

Change-Id: I53c81b6fa0d42edd211499e15fcecade53328eac

Flamefire committed:
XperiaParts: Factor out OpenCS start activity intent creation

Avoid repeating the package name

Change-Id: Ia0834d1381931bec2ef53db93475349a943c29c1

Flamefire committed:
OpenCS: Unify Intent creation style

Don't create the intent in the `start*` call but as a separate variable
with indented options

Change-Id: Id0f357d485e1e5d05f37c53896506864353a1151

Flamefire committed:
OpenCS: Simplify preference handling

Create the `SharedPreferences` for `PREF_PKG` fully in `Configurator`
such that device protected storage is consistently handled.

Change-Id: I9ab9a02332ed497da1ee839a9330b5a8e07bb340

Flamefire committed:
OpenCS: Remove braces around single-line conditions

Less vertical space used and consistent now.

Change-Id: I0afeb2143d4a4a555556325d3a9d9f7135a1c2fb

Flamefire commented:
> Yeah, overall nice changes. Makes the code much clearer, especially in some fixed int cases, like `addFlags(270565376)`. Really good work!

Thanks. That constant was already resolved in one other place, just needed to double check and adapt.

I had a few minor issues that failed the build (e.g. `final` can't be used if the values are only assigned in `Service::onCreate`) that I now fixed. Also found a few more unnamed variables such as `String str`

If you could give it a final check? I compile tested it at least, so should be fine.

antoniu200 commented:
Finally got time to do a last relatively quick check.
It looks alright, logic does not seem to have been altered overall.
As for OOP changes, I can barely judge, given my OOP knowledge is very little.

My opinion is the changes are welcome, they make the code far less abstract and easier to understand. As of now, I cannot do proper testing, since my XZ1 Compact's USB port just died, will replace it hopefully next week.
